### PR TITLE
Add PR curve to ptg-eval

### PR DESCRIPTION
**Pull request by hdefazio**

### CODE REVIEW

**Comment by hdefazio** [comment]
@Purg PTAL

File: angel_system/eval/ptg_evaluate.py [hunk 1]
```diff
@@ -74,17 +98,16 @@ def run_eval(args):
...
     # Plot
     # ============================
-    for label, ts_range_pairs in gt_label_to_ts_ranges.items():
-        if label in dets_label_to_ts_ranges:
-            plot_activity_confidence(label=label, gt_ranges=ts_range_pairs, det_ranges=dets_label_to_ts_ranges, output_dir=output_dir)
-        else:
-            log.warning(f"No detections found for \"{label}\"")
+    vis = EvalVisualization(labels=labels, gt=gt_label_to_ts_ranges, dets=dets_label_to_ts_ranges, output_dir=output_dir)
+    vis.plot_activities_confidence()
+    vis.plot_pr_curve()
```
Comment (Purg) [inline angel_system/eval/ptg_evaluate.py:103]: I presume we want the intersection threshold to be higher than 0.0 (which is currently the default parameter value for `plot_pr_curve)?

File: angel_system/eval/visualization.py [hunk 2]
```diff
@@ -4,74 +4,161 @@
...
+            id = row['id']
+            label = row['class']
+
+            truth = [1 if det['iou'] > iou_thr else 0 for det in self.dets[label]]
+            pred = [det['conf'] for det in self.dets[label]]
+
+            precision[i], recall[i], thresholds[i] = precision_recall_curve(truth, pred)
+            average_precision[i] = average_precision_score(truth, pred)
+
+            display = PrecisionRecallDisplay(
```
Comment (Purg) [inline angel_system/eval/visualization.py:148]: Usage of this class is new to me! Is this to provide a convenient plot formatting with the axis and legend like in the scikit-learn docs page?

In the docs page, it does say that [`from_predictions`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.PrecisionRecallDisplay.html#sklearn.metrics.PrecisionRecallDisplay.from_predictions) is recommended instead of directly creating this class. It might be simpler to call that here instead of calling `precision_recall_curve` and `average_precision_score` manually (it calls those methods for you).

File: angel_system/eval/compute_scores.py [hunk 3]
```diff
@@ -8,21 +8,23 @@ def iou_per_activity_label(labels, gt, dets):
     """
     Calculate the iou per activity label
```
[the remark this answers is not in the document]
Comment (hdefazio) [reply angel_system/eval/compute_scores.py:9]: I've updated the iou -> detection intersection in  #107 . I figured I could land that after this MR, but if it makes it easier I can merge that into this branch.

File: angel_system/eval/compute_scores.py [hunk 4]
```diff
@@ -8,21 +8,23 @@ def iou_per_activity_label(labels, gt, dets):
...
 
-    :param labels: List of String labels of the activity classes
+    :param labels: Pandas df with columns id (int) and class (str)
     :param gt: Dict of activity start and end time ground truth values, organized by label keys
     :param dets: Dict of activity start and end time detections with confidence values, organized by label keys
 
-    :return: Tuple(Average IoU across all classes, Dictionary mapping class labels to their average IoU scores)
+    :return: Tuple(Average IoU across all classes, Dictionary mapping class labels to their average IoU scores, dets)
```
[the remark this answers is not in the document]
Comment (hdefazio) [reply angel_system/eval/compute_scores.py:15]: I was returning the dets back out so the visualization code wouldn't have to repeat the "detection intersection" calculations. I could just call the detection intersection calculations again from the visualization code (which would help ensure that dets actually has the "detection intersection" set instead of assuming that this was already called)? I will add the note about modifying in-place though

File: pyproject.toml [hunk 5]
```diff
@@ -20,6 +20,9 @@ timm = "^0.5.4"
 einops = "^0.4.1"
 pynput = "^1.7.6"
 pandas = "^1.4.3"
+sklearn = "^0.0"
+pyarrow = "^9.0.0"
```
[the remark this answers is not in the document]
Comment (hdefazio) [reply pyproject.toml:24]: I believe it is a dependency of sklearn (or it may have been pandas...)? I get dependency errors without it.

[the remark this answers is not in the document]
Comment (Purg) [reply pyproject.toml:23]: pyarrow doesn't seem to get installed if I install just pandas or scikit-learn. If you get that dep error you mentioned again I would be interested in seeing it.

File: angel_system/eval/compute_scores.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Purg) [reply angel_system/eval/compute_scores.py:9]: Like we discussed at stand-up, might as well just merge that into here if it is just the naming update.

File: angel_system/eval/compute_scores.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Purg) [reply angel_system/eval/compute_scores.py:16]: You are correct in that the visualization code should not repeat the detection intersection again. I would say remove the `dets` return from here. This function modifies the input data structure, which is likely sufficient. Just add comments enough to describe what this function is doing to the `dets` structure.

File: angel_system/eval/visualization.py [hunk 6]
```diff
@@ -4,74 +4,150 @@
...
+        ax.set_title("Precision vs. Recall")
+
+        colors = plt.cm.rainbow(np.linspace(0, 1, len(self.labels)))
+        
+        # ============================
+        # Add F1 score 
+        # ============================
+        fscores = np.linspace(0.2, 0.8, num=4)
+        for f_score in fscores:
+            x = np.linspace(0.01, 1)
```
Comment (Purg) [inline angel_system/eval/visualization.py:122]: The plot as-is shows the f1=0.2 line to have a floating start:
PR

`0.001` results in a completed topographic line for the area covered. 
PR
```suggestion
            x = np.linspace(0.001, 1)
```

**Review by Purg — COMMENTED** [review]
Minor plot nitpick, otherwise LGTM

**Review by Purg — APPROVED** [review]
